### ISSUE TITLE
Phase vcf with pre phased reads

### ISSUE THREAD

**Issue by mrvollger** [PacificBiosciences/HiPhase#30]
Hi @holtjma,

It's on my todo list to write a tool that takes a bam with phased reads and then phases a vcf with that preexisting information. This often comes up for us when we phase reads with k-mers or an assembly and we want to create a phased vcf to match. 

I was going to make a standalone tool but then I thought depending on how things are setup in hiphase much of the work might already be done. 

Do you think adding a `--use-pre-phased-bam` would be something you'd allow? If so, could you maybe point me to a place within the code where I might be able to start working in this functionality? 

Cheers,
Mitchell

**Comment by holtjma** [comment]
Hey Mitchell,

I want to make sure I understand before digging into this too much.  Is this basically a reverse-haplotagging step? I.e. the BAM is already tagged with haplotype (HP:1 or HP:2) and phase set information (PS tag) but the VCF is unphased for some reason?  Or is it something different?

Matt

**Comment by mrvollger** [comment]
Yep, that's right. The bam already has H1/H2 tags (not PS tags because, in these cases, I have chromosome length phasing, but I could hack in arbitrary PS blocks), and I have an unphased vcf. 

This comes up when I do reference free phasing of reads via k-mers or a de-novo assembly, and then I want to have consisting phasing of a vcf and the bam file after I e.g. realign to GRCh38. 

I'd understand if there wasn't a good place to hook into hiphase for this, but I figured Id ask.

**Comment by holtjma** [comment]
Hey @mrvollger, just revisiting the relevant code blocks this morning.  As I mentioned before, the current framework won't let you do this, it's basically expecting to phase everything at once (BAM & VCF).  I do (at a high level), see a path where it could be modified to make this happen, but it doesn't seem like trivial changes. 

Because of that, I don't think it's going to prioritized in the near future. I made a note of the possible extension, but if this is something you desire soon, I would recommend going forward with some self-built tooling (I'm also unaware of any tools that currently do what you're asking). Sorry, this probably isn't the answer you were hoping for!

**Comment by mrvollger** [comment]
@holtjma,

All good, I expected as much. Looks like the whatshap devs are open to it and Sergey Nurk's team has already done some ground work, so I am going to help make it happen in whatshap. 

Cheers,
Mitchell
